docs/src/modules/concepts/pages/ms-agent-patterns.adoc

```diff
@@ -0,0 +1,143 @@
+* Agent interaction is more like collaboration than sequential hand-offs
+
+=== Examples
+
+* In xref:getting-started:planner-agent/team.adoc[this example], this pattern is illustrated well with a workflow with deterministic steps (no dynamic planning) that calls agents in sequence
+
+
+== Concurrent orchestration
+
+https://learn.microsoft.com/en-us/azure/architecture/ai-ml/guide/ai-agent-design-patterns#concurrent-orchestration[Concurrent orchestration, window="new"] refers to running multiple AI agents simultaneously working on the same task. The outputs of all concurrent agents are then collected and processed. This is ideal when you have a number of agentic tasks that do not rely on the outputs of others.
+
+In Microsoft’s original diagram, there is an “Initiator and collector agent.” In the Akka version, we’ve simply replaced the collector agent with a workflow. Nearly everything else remains the same.
+
+image:ai_arch_concurrent.jpg[Diagram showing concurrent workflow execution in agentic app]
+
+Also note that in this diagram, the workflow is responsible for controlling `agent 1.1` and `agent 1.2`, which appear as sub-agents in Microsoft’s pattern. In keeping with the golden rule of agentic composition, Akka agents don’t spawn sub-agents, workflows decide which agents are needed and the Akka runtime takes care of provisioning.
+
+As you’ll see later in this document, Akka workflows can easily spawn sub-workflows. This reinforces the notion that the only real difference between these patterns in Akka and elsewhere is that Akka separates the roles of orchestration and model communication while most other frameworks choose to combine them.
+
+More advanced concurrent orchestration could be implemented by a parent workflow spawning child workflows. In this pattern, each child workflow performs a multi-step task and then delivers the result back to the parent workflow as a message (i.e. method call). 
+The parent workflow pauses when waiting for the results from children. The results would be stored in the state, and when the parent workflow is satisfied with all of the collected results it transitions to another step. 
+
+In this kind of advanced scenario, Akka takes care of all the hard parts like managing distributed state, distributed long-running timers, workflow resiliency, and much more.
+
+=== Examples
+The following code snippet shows how easily we can create a workflow step that calls two agents concurrently and gathers their results to be passed to the next step in the workflow:
+
+[source,java,indent=0]
+.ConcurrentWorkflow.java
+----
+include::java:example$doc-snippets/src/main/java/patterns/ConcurrentWorkflow.java[tag=concurrent-step]
+----
+<1> Call the `WeatherAgent` and `TrafficAgent` with `invokeAsync`, which immediately returns a `CompletionStage` of the future result.
+<2> Collect the results from the two agents.
+<3> Update the workflow state and continue with next step.
```

Comment:
I extracted the example to compiled code snippet: https://github.com/akka/akka-sdk/pull/913/commits/8e410348e22f2156d41c1a1ce431ba12545787a0

Note that I'm using toCompletionStage.get so far, since the await sugar doesn't exist yet. We'll discuss next week if it's needed, when the others are back.